esx_service/vmdk_ops.py

```diff
@@ -862,83 +874,114 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts, vc_u
     Returns None (if all OK) or error string
     """
     logging.debug("config_path=%s", config_path)
+    # get datastore the VM is running on
     vm_datastore_url = vmdk_utils.get_datastore_url_from_config_path(config_path)
     vm_datastore = get_datastore_name(vm_datastore_url)
     logging.debug("executeRequest: vm_datastore = %s, vm_datastore_url = %s",
                   vm_datastore, vm_datastore_url)
 
     error_info, tenant_uuid, tenant_name = auth.get_tenant(vm_uuid)
+    force_detach = False
     if error_info:
-        # For docker volume ls, docker prints a list of cached volume names in case
-        # of error(in this case, orphan VM). See Issue #990
-        # Explicity providing empty list of volumes to avoid misleading output.
+        # For "docker volume ls", in case of error fron the plugin Docker prints a list of cached volume names,
+        # which is misleading. To avoid this, we replace error with an empty list. See Issue #990 for details.
         if (cmd == "list") and (not tenant_uuid):
             return []
+        # We need special handling for failure to find tenants in "detach".
+        # get_tenant() will  fail if the VM was removed from the vmgroup or the vmgroup was deleted
```

Comment:
If it is a user defined vmgroup, deleting the vmgroup to which VM belongs or removing it from the vmgroup cannot complete if VM has disk attached. But for default vmgroup this doesn't hold true because we don't have list of VMs belonging to default vmgroup.
We can be specfic about this in the comment